game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java

```diff
@@ -95,21 +234,34 @@ public AggregateResults calculate(
       final Collection<Unit> bombarding,
       final Collection<TerritoryEffect> territoryEffects,
       final boolean retreatWhenOnlyAirLeft,
-      final int runCount)
+      final int initialRunCount)
       throws IllegalStateException {
-    Preconditions.checkState(!isShutDown, "ConcurrentBattleCalculator is already shut down");
-    Preconditions.checkState(bytes.length != 0, "Data has not been set yet.");
-    final long start = System.currentTimeMillis();
-    final int runsPerWorker = runCount / MAX_THREADS;
-    final List<Future<AggregateResults>> results;
-    synchronized (mutex) {
-      results =
-          IntStream.range(0, MAX_THREADS)
-              .map(index -> index == 0 ? runCount % MAX_THREADS : 0)
-              .map(runs -> runs + runsPerWorker)
-              .mapToObj(
-                  individualRemaining ->
-                      createBattleCalcWorker(
+    synchronized (mutexCalcIsRunning) {
+      awaitLatch();
+      final long start = System.currentTimeMillis();
+      // Create worker thread pool and start all workers
+      int totalRunCount = 0;
+      int runCount = initialRunCount;
+      final int workerNum = workers.size();
+      final int workerRunCount = Math.max(1, (runCount / Math.max(1, workerNum)));
```

Comment:
This code here changed slightly from the original implementation, basically we compute all of the run counts here now.
Basically it does the same as my IntStream implementation to the left with some added exit conditions